Maple2.Server.Game/Manager/UgcMarketManager.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Maple2.Database.Extensions;
+using Maple2.Database.Storage;
+using Maple2.Model.Enum;
+using Maple2.Model.Game;
+using Maple2.Model.Metadata;
+using Maple2.Server.Game.Packets;
+using Maple2.Server.Game.Session;
+using Serilog;
+
+namespace Maple2.Server.Game.Manager;
+
+public sealed class UgcMarketManager {
+    private readonly GameSession session;
+
+    private readonly IDictionary<long, UgcMarketItem> items;
+    private IDictionary<long, SoldUgcMarketItem> sales;
+    private IDictionary<long, PlayerInfo> favoriteDesigners;
+
+    private readonly ILogger logger = Log.Logger.ForContext<UgcMarketManager>();
+
+    public UgcMarketManager(GameSession session) {
+        this.session = session;
+
+        using GameStorage.Request db = session.GameStorage.Context();
+        items = db.GetUgcListingsByAccountId(session.AccountId);
+        sales = db.GetMySoldUgcListings(session.AccountId);
+        favoriteDesigners = new Dictionary<long, PlayerInfo>();
+        IList<long> designerList = session.Config.GetFavoriteDesigners();
+
+        foreach (long characterId in designerList) {
+            if (!session.PlayerInfo.GetOrFetch(characterId, out PlayerInfo? playerInfo)) {
+                continue;
+            }
+            favoriteDesigners.Add(characterId, playerInfo);
+        }
+    }
+
+    public void Load() {
+        session.Send(MeretMarketPacket.LoadPersonalListings(items.Values));
+    }
+
+    public void RefreshSales() {
+        using GameStorage.Request db = session.GameStorage.Context();
+        Save(db);
+        sales.Clear();
+        sales = db.GetMySoldUgcListings(session.AccountId);
+        session.Send(MeretMarketPacket.LoadSales(sales.Values));
+    }
+
+    public void Add(UgcMarketItem item) {
```

Comment:
`ListItem` might be a better name than `Add` since it's the counterpart of `UnlistItem`?